### PR TITLE
Docs: Code contribution WSL removal

### PR DESCRIPTION
**Pull request by tobiasdiez**
Closes #

## What I did

After the recent changes (in particular the usage of jiti), I have no problems developing on storybook directly from windows, without any need for WSL. Thus this is removed from the developer docs. 
However, one does need to run the command for creating the sandbox as admin, since otherwise one gets a permission error when creating the symlinks.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

## Greptile Summary

This PR removes the requirement for Windows users to use Windows Subsystem for Linux (WSL) when developing Storybook, simplifying the setup process.

- Updated CONTRIBUTING.md to remove WSL requirement for Windows users
- Modified docs/contribute/code.mdx to recommend using a terminal with admin privileges instead of WSL
- Advised running sandbox creation command as admin to avoid permission errors with symlinks
- Simplified setup process may have security implications due to elevated privileges

### CODE REVIEW

File: docs/contribute/code.mdx [hunk 1]
```diff
@@ -10,7 +10,7 @@ Contribute a new feature or bug fix to [Storybook's monorepo](https://github.com
 ## Prerequisites
 
 * Ensure you have Node version 18 installed (suggestion: v18.16.0).
-* Ensure if you are using Windows to use the Windows Subsystem for Linux (WSL).
+* Ensure if you are using Windows to run the commands below in a terminal with administrator privileges.
```
Comment (jonniebigodes) [inline docs/contribute/code.mdx:13]:
```suggestion
* If you're working with Windows, all commands should be run in a terminal with administrator privileges.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -20,7 +21,7 @@ You will need to have the following installed:
 
 ## Running the local development environment
 
-- Ensure if you are using Windows to use the Windows Subsystem for Linux (WSL).
+- Ensure if you are using Windows to use a terminal with admin privileges.
```
Comment (jonniebigodes) [inline CONTRIBUTING.md:24]:
```suggestion
- All commands should be run in a terminal with administrator privileges in Windows environments.
```

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@tobiasdiez thanks for taking the time to put together this pull request and helping us improve the documentation with the recent changes introduced. Appreciate it 🙏 .

This is shaping up nicely, I left only a couple of small adjustments for you to consider when you can. 

Let me know and we'll take it from there.

Have a great day.

Stay safe

**Comment by tobiasdiez** [comment]
Thanks @jonniebigodes. Your suggestions are a nice improvement.

**Review by jonniebigodes — APPROVED** [review]

**Comment by jonniebigodes** [comment]
@tobiasdiez I'm sorry for the delayed response and for getting this across, but it's been some busy times here at Storybook. I've just updated the branch and am going to merge this one, as you already addressed the feedback. 

Have a great day.

Stay safe

**Pull request merged**